LeaderboardBackend/Services/Impl/RunService.cs

```diff
@@ -1,17 +1,89 @@
+using System.Text.Json;
+using LeaderboardBackend.Models;
 using LeaderboardBackend.Models.Entities;
+using LeaderboardBackend.Models.Requests;
+using LeaderboardBackend.Result;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using OneOf.Types;
 
 namespace LeaderboardBackend.Services;
 
-public class RunService(ApplicationContext applicationContext) : IRunService
+public class RunService(ApplicationContext applicationContext, IOptions<JsonOptions> jsonOptions) : IRunService
 {
     public async Task<Run?> GetRun(Guid id) =>
         await applicationContext.Runs.Include(run => run.Category).SingleOrDefaultAsync(run => run.Id == id);
 
-    public async Task CreateRun(Run run)
+    public async Task<CreateRunResult> CreateRun(User user, long categoryId, JsonDocument request)
```

Comment:
As you can see, I've pushed the responsibility of parsing the request body to the service itself. Alternatively, I could create a parser class in a separate file, or a parser method in the controller file instead, which I imagine will return a `OneOf<CreateTimedRunRequest, CreateScoredRunRequest>`, and in turn be passed to the service.

Reply:
This is irrelevant, now.